AK/Optional.h

```diff
@@ -311,7 +311,10 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
     }
 
 private:
-    alignas(T) u8 m_storage[sizeof(T)];
+    union {
+        char m_null;
+        T m_storage;
+    };
```

Comment:
I think this technically is a breaking change before the `addressof` one

Reply:
Indeed. Should I change it?

Reply:
Seeing that we don't overload `operator&`[^1] - no need,
It may even prompt dropping the `addressof` part all together, but a maintainers word on that'd be the deciding factor

[^1]: And likely will stop anyone from doing so

Reply:
(I guess I count as a maintainer)

Agreed with @Hendiadyoin1 on removing `addressof`. We generally try to favor a readable/simple implementation over covering every edge case (if you overload the address-of operator, you're asking for trouble and the "Requested Changes" label on GitHub).

Reply:
I've removed `addressof`